src/libguac/client.c

```diff
@@ -185,6 +185,12 @@ static void guac_client_promote_pending_users(union sigval data) {
     /* Acquire the lock for reading and modifying the list of pending users */
     guac_rwlock_acquire_write_lock(&(client->__pending_users_lock));
 
+    /* Skip user promotion entirely if there's no pending users */
+    if (client->__pending_users == NULL) {
+        guac_rwlock_release_lock(&(client->__pending_users_lock));
```

Comment:
Any reason to not move the `promotion_complete` label up a line and rely on the `guac_rwlock_release_lock()` call that already exists there?

Reply:
Nope!